corehq/tabs/uitab.py

```diff
@@ -127,7 +124,8 @@ def should_show(self):
         if not self.show_by_default and not self.is_active_tab:
             return False
 
-        if self.restrict_access_by_location and not self.filtered_dropdown_items:
```

Comment:
I think you're right, there must be something excluding the menu bar from the invitation page, but I introduced this reference.  Now we only call `self.filtered_dropdown_items` when it's for sure a restricted user, which means there is a domain and the user is a member of the domain with a restricted role.